src/runtime/particle-execution-context.ts

```diff
@@ -385,30 +357,16 @@ export class ParticleExecutionContext implements StorageCommunicationEndpointPro
       canWrite: connSpec.isOutput,
     });
     handleMap.set(name, handle);
-
-    // Defer registration of handles with proxies until after particles have a chance to
-    // configure them in setHandles.
-    registerList.push({proxy, particle, handle});
```

Comment:
should we support this in the new stack as well? I think at the moment you can change options only after the handle is already registered with the proxy

Reply:
We should support changes to the options at any time; this was a hack because we ignored changes after setHandles was called. So we should redesign and add an appropriate mechanism rather than preserving this.